Add more specs for YIP-0001 (#120)
layer-3/clearsync: docs/YIP/YIP-0001-price-precision.md

## Decision

### Definitions

**Price** - asset price in quote currency.

**Quantity** - in trading, usually refers to the number of units or shares
of a particular security or asset that is being bought or sold.
For instance, if an investor buys 100 shares of a company's stock, the quantity in this case is 100.

| Number      | Round to |
| ----------- | ----------- |
| Price       | 18       |
| Quantity    | 8        |

> Significant figures, also known as the precision of a number in positional notation, are digits in the number that are reliable and necessary to indicate the quantity of something.

Reviewer: ```suggestion
**Significant figures** (also known as the precision of a number in positional notation) - are digits in the number that are reliable and necessary to indicate the quantity of something.
```

> The precision level of all trading prices is calculated based on significant figures. Yellow members uses five (5) significant digits for the prices of all pairs of the network.

Reviewer: ```suggestion
The precision level of all trading prices is calculated based on significant figures. Yellow members shall use five (5) significant digits for the prices of all pairs of the network.
```

> Alternative options for significant figures is in range of 5 - 8.

Reviewer: ```suggestion
Alternative options for significant figures are in range of 5 - 8.
```

> Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.

Reviewer: ```suggestion
Some examples of _five significant digits_ are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
```

This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the quantity increases. The opposite is true for minimal amounts, where greater precision is more valuable.

Quantity precision allows up to 8 decimals. Anything exceeding this shall be rounded to the 8th decimal.

## Consequences

1. All API outputting prices shall be truncated and formation using five (5) significant figures

> 3. All API input shall return an error for orders where prices or amount are lower in precision

Reviewer 2: All API input shall return an error for orders where prices or amounts exceed 18 and 8 precision respectfully

## Decision

### Definitions

**Price** - asset price in quote currency.

> **Quantity** - in trading, quantity usually refers to the number of units or shares

Reviewer 2: The second `quantity` is excessive

of a particular security or asset that is being bought or sold.
For instance, if an investor buys 100 shares of a company's stock, the quantity in this case is 100.

**Value** - stands for USD market value, value is always in USD (for example for market BTC/ETH value will be in USD).

| Number      | Round to |
| ----------- | ----------- |
| Price       | 18       |
| Quantity    | 8        |

> | Value       | 2        |

Reviewer 2: Add an explanation about what this table is.

Author: we dropped value at this point

Significant figures, also known as the precision of a number in positional notation, are digits in the number that are reliable and necessary to indicate the quantity of something.

The precision level of all trading prices is calculated based on significant figures. Yellow members uses five (5) significant digits for the prices of all pairs of the network.
Alternative options for significant figures is in range of 5 - 8.

Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.

This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the quantity increases. The opposite is true for minimal amounts, where greater precision is more valuable.

> Quantity precision allows up to 8 decimals. Anything exceeding this will be rounded to the 8th decimal.

Reviewer 2: Change `will` to `shall` according to meta YIP

Author: fixed

> ## Consequences

Reviewer 2: What is consensus for other parameters like Quantity and Value?

Author: we didn't have consensus on value param and we don't need it at this point

> 1. All API outputting prices should be truncated and formation using five (5) significant figures

Reviewer 2: Change  `should` to `shall` according to meta YIP

> 3. All API input will return an error for orders where prices are lower in precision

Reviewer 2: `shall` again. 
Is it possible to define the error format and message at the YIP stage?

Author: I'm not ready to define final error right now

> 4. All frontend and UI clients should truncate order price input to five (5) significant figures

Reviewer 2: Change `should` to `shall` according to meta YIP

> 5. Backend and UI pkgs should support easy change of significant figures

Reviewer 2: Change `should` to `shall` according to meta YIP.
Define the limits of change, like from 5 to 8 including 5 and 8.

Significant figures, also known as the precision of a number in positional notation, are digits in the number that are reliable and necessary to indicate the quantity of something.

The precision level of all trading prices is calculated based on significant figures. Yellow members uses seven (7) significant digits for the prices of all pairs of the network.
Alternative options for significant figures is in range of 5 - 8.

> Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340 those examples for 5 significant figures.

Reviewer: 'five significant digits' is repeated twice

This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the amount increases. The opposite is true for minimal amounts, where greater precision is more valuable.

> Amount and price precision allows up to 8 decimals. Anything exceeding this will be rounded to the 8th decimal.

Author: price - 10 decimal precision, 5 significant figures

## Consequences

1. All API outputting prices should be truncated and formation using seven (7) significant figures

> 2. All API outputting prices and amounts allows up to truncated 8 decimals. Anything exceeding this will be rounded to the 8th decimal.

Author: API level - return error
UI - will not allow to input
